nat-instance.tf

```
@@ -1,27 +1,28 @@
 module "nat_instance_label" {
+  enabled    = var.enabled
   source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.16.0"
   context    = module.label.context
   attributes = distinct(compact(concat(module.label.attributes, ["nat", "instance"])))
 }
 
 locals {
   cidr_block               = var.cidr_block != "" ? var.cidr_block : join("", data.aws_vpc.default.*.cidr_block)
-  nat_instance_enabled     = var.enabled && var.nat_instance_enabled ? 1 : 0
+  nat_instance_enabled     = var.nat_instance_enabled ? 1 : 0
   nat_instance_count       = var.nat_instance_enabled && ! local.use_existing_eips ? length(var.availability_zones) : 0
   nat_instance_eip_count   = local.use_existing_eips ? 0 : local.nat_instance_count
   instance_eip_allocations = local.use_existing_eips ? data.aws_eip.nat_ips.*.id : aws_eip.nat_instance.*.id
 }
 
 resource "aws_security_group" "nat_instance" {
-  count       = local.nat_instance_enabled
+  count       = var.enabled ? local.nat_instance_enabled : 0
```

This would be much more readable and consistent than using the current local variable.

```suggestion
  count       = var.enabled && var.nat_instance_enabled ? 1 : 0
```